pyads/pyads_ex.py

```diff
@@ -176,6 +177,34 @@ def adsAddRoute(net_id: SAmsNetId, ip_address: str) -> None:
         raise ADSError(error_code)
 
 
+def send_raw_udp_message(
+    ip_address: str, message: bytes, expected_return_length: int
+) -> Tuple[bytes, str]:
+    """Send a raw UDP message to the PLC and return the response
+
+    :param str ip_address: ip address of the PLC
+    :param message: the message to send to the PLC
+    :expected_return_length: number of bytes to expect in response
+    """
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_DGRAM)) as sock:  # UDP
+        # Listen on any available port for the response from the PLC
+        sock.bind(("", 0))
+
+        # Send our data to 48899 on the PLC, which is where ADS expects UDP
+        # messages
+        sock.sendto(message, (ip_address, 48899))
```

Comment:
We should not have hard-coded ports in the code. I think it would be nice to put this port number into [constants.py](https://github.com/stlehmann/pyads/blob/440d0a18493000c407cc2e65a1c795902948ac36/pyads/constants.py#L189) in the PORTS section. Something like PORT_REMOTE_UDP.